ext/standard/var.c

```diff
@@ -671,7 +671,9 @@ static inline zend_long php_add_var_hash(php_serialize_data_t data, zval *var, b
 		return 0;
 	} else if (!in_rcn_array
 	 && Z_REFCOUNT_P(var) == 1
-	 && (Z_OBJ_P(var)->properties == NULL || GC_REFCOUNT(Z_OBJ_P(var)->properties) == 1)) {
+	 && (Z_OBJ_P(var)->properties == NULL || GC_REFCOUNT(Z_OBJ_P(var)->properties) == 1)
+	 /* __serialize may arbitrarily increase the refcount */
+	 && Z_OBJCE_P(var)->__serialize == NULL) {
```

Comment:
Isn't this also a possible problem for ``__sleep()``?

Reply:
Indeed, added a check for this and a test. Although the check is unfortunately more expensive than the one for __serialize, so we maybe need a different solution...

Reply:
So another follow-up question is, what happens if [Serializable](https://www.php.net/manual/en/class.serializable.php) is used without ``__serialize()`` (other than a deprecation notice)?

We really should think about deprecating ``__sleep()``/``__wakeup()`` as it seems to me the functionality is covered by the new serialize magic methods.

Reply:
> So another follow-up question is, what happens if [Serializable](https://www.php.net/manual/en/class.serializable.php) is used without __serialize() (other than a deprecation notice)?

I don't think this is a problem for this bug? But I could be wrong or misunderstanding the question.

Reply:
IIRC If ``Serializable`` is implemented, ``serialize()`` function will call the ``serialize()`` method of the object, similar to ``__sleep()``/``__serialize()``

Reply:
That serialize methods is supposed to returns the string that will be embedded in serialize's output. As the bug manifests in the built-in reference tracking when it performs the default serialization, I don't think we'll have that issue in that case?

Reply:
Ah so that's how it works, yeah should be fine then